### PR TITLE
chore: [UM-11] add a new functionality to call bulk users retrieve

### PR DESCRIPTION
**Pull request by ppolito-zextras**

### CODE REVIEW

File: carbonio-user-management-sdk.iml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (nomanAli95) [inline carbonio-user-management-sdk.iml]: Please remember to not commit or push this kind of file, this is a module settings file created by IntelliJ, so remove it from this PR and add a line in .gitignore for the iml extension

File: pom.xml.versionsBackup [hunk 2]
```diff
@@ -0,0 +1,83 @@
+<!--
```
Comment (nomanAli95) [inline pom.xml.versionsBackup]: same as above, please check and remove this from the PR

File: src/main/java/com/zextras/carbonio/usermanagement/UserManagementClient.java [hunk 3]
```diff
@@ -106,6 +109,47 @@ public Try<UserId> validateUserToken(String carbonioUserToken) {
...
+   * {@link Try#failure} containing an {@link InternalServerError} throwable if something goes
+   * wrong during the API call.
+   */
+  public Try<List<UserInfo>> getUsers(
+    String cookie,
+    List<UUID> usersUUID
+  ) {
+    CloseableHttpClient httpClient = HttpClients.createMinimal();
+
+    HttpGet request = new HttpGet(userManagementURL + getUsersEndpoint + usersUUID);
```
Comment (nomanAli95) [inline src/main/java/com/zextras/carbonio/usermanagement/UserManagementClient.java]: Please check usersUUID, I don't think you can pass it in this way. The correct way to pass a list of ids in a GET http request is: `/user/?ids=1&ids=2&ids=3`

**Review by nomanAli95 — CHANGES_REQUESTED** [review]

File: src/main/java/com/zextras/carbonio/usermanagement/UserManagementClient.java [hunk 4]
```diff
@@ -106,6 +110,49 @@ public Try<UserId> validateUserToken(String carbonioUserToken) {
...
+   */
+  public Try<List<UserInfo>> getUsers(
+    String cookie,
+    List<UUID> usersUUID
+  ) {
+    CloseableHttpClient httpClient = HttpClients.createMinimal();
+
+    AtomicReference<String> getRequest = new AtomicReference<>(userManagementURL + getUsersEndpoint + "?userIds=");
+    usersUUID.forEach(userUUID -> getRequest.updateAndGet(v -> v + userUUID));
+    HttpGet request = new HttpGet(getRequest.get());
```
Comment (lucagasparini1981) [inline src/main/java/com/zextras/carbonio/usermanagement/UserManagementClient.java]: Request URL is not valid. You must have an URL like this: 
`http://localhost:10000/users?userIds={id1}&userIds={id2} ...`
You can use usersUUID.stream().map(...).collect(Collectors.joining(...)) to build the string of queries' parameters.

File: carbonio-user-management-sdk.iml (hunk 1 above)
Comment (lucagasparini1981) [reply carbonio-user-management-sdk.iml]: Please remove this file from the repository

File: src/main/java/com/zextras/carbonio/usermanagement/UserManagementClient.java [hunk 5]
```diff
@@ -106,6 +110,49 @@ public Try<UserId> validateUserToken(String carbonioUserToken) {
...
+        String bodyResponse = IOUtils.toString(
+          response.getEntity().getContent(),
+          StandardCharsets.UTF_8
+        );
+
+        return Try.success(new ObjectMapper().readValue(bodyResponse, new TypeReference<>() {}));
+      }
+
+      return Try.failure(
+        new UserNotFound("The cookie: \"" + cookie + "\" is not valid")
```
Comment (nomanAli95) [inline src/main/java/com/zextras/carbonio/usermanagement/UserManagementClient.java]: I think this is not the correct error you should return here, it depends on how you handle this request on the user-management side. Infact I noticed that the yaml on the carbonio-user-management side has only 400 bad request error, so here you couldn't use "The cookie is not valid" as message error.

**Review by nomanAli95 — DISMISSED** [review]

**Review by lucagasparini1981 — CHANGES_REQUESTED** [review]

File: src/main/java/com/zextras/carbonio/usermanagement/UserManagementClient.java [hunk 6]
```diff
@@ -106,6 +111,48 @@ public Try<UserId> validateUserToken(String carbonioUserToken) {
...
```
Comment (domenicostragliotto) [inline src/main/java/com/zextras/carbonio/usermanagement/UserManagementClient.java]: The Exception message is not coherent with the exception

**Review by domenicostragliotto — CHANGES_REQUESTED** [review]

**Review by domenicostragliotto — DISMISSED** [review]

**Review by domenicostragliotto — DISMISSED** [review]

**Review by federicorispo — APPROVED** [review]

**Review by lucagasparini1981 — APPROVED** [review]

**Review by lucagasparini1981 — DISMISSED** [review]

**Review by federicorispo — DISMISSED** [review]

**Review by federicorispo — APPROVED** [review]

**Review by domenicostragliotto — APPROVED** [review]

**Review by lucagasparini1981 — APPROVED** [review]

**Pull request merged**